### PR TITLE
Fix tabs behaviour

### LINKED ISSUE #5: needs a method to set tab positions
```
What steps will reproduce the problem?
try and do a horizontal tab

What is the expected output? What do you see instead?
no output

What version of the product are you using? On what operating system?


Please provide any additional information below.

the horizontal and I think vertical tabs do nothing (or might do a line feed) 
until the tab positions have been set with ESC D <list of columns> null
```

Original issue reported on code.google.com by `alanbell...@gmail.com` on 15 Nov 2012 at 10:40

### PR DESCRIPTION
**Pull request by csoft2k**
### Contributor checklist

- [x] I have read the CONTRIBUTING.rst
- [x] I have tested my contribution on these devices: Epson TM-T20II
- [x] My contribution is ready to be merged as is

----------

### Description
The changes done in this commit should help with the open issues:
#5, #27 and #161.
The old implementation lacked the NUL char at the end of the command, as
defined on the Epson ESC/POS Reference Guide (see
https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=53
). Also, the horizontal tab control character (CTL_HT) shouldn't be
there.

This implementation allows setting up to 32 tabs with a given tab width.
Both values are checked to be in the valid ranges defined on the guide.
Also, the TabPosError exception text has been rewritten to define the
stated above.

fixes #5

### CODE REVIEW

**Comment by csoft2k** [comment]
Seems I created a pull request with all my old commits. Let me know if this is a problem.

**Comment by csoft2k** [comment]
Please find below 2 examples using 2 and 8 chars-width tabs:
2 examples

**Comment by patkan** [comment]
Thank you! If you do not know how to rebase this in order to remove your old posts I can simply do it while merging.
The change looks good to me at first glance. I will review it asap.

**Comment by patkan** [comment]
As a hint: to avoid problems like these in the future (this also prevents that you have to have merge-commits in your branch) you should develop new features always in an extra branch. This way you can simply track the upstream development branch.

**Comment by csoft2k** [comment]
Seems I've managed to mix thing a bit more. Please integrate just [this commit](https://github.com/python-escpos/python-escpos/pull/238/commits/ee29d42a5ccfe9b2fd01208ed1eedc5ee8f2eba9).

File: src/escpos/escpos.py [hunk 1]
```diff
@@ -711,13 +712,14 @@ def control(self, ctl, pos=4):
         elif ctl.upper() == "CR":
             self._raw(CTL_CR)
         elif ctl.upper() == "HT":
-            if not (1 <= pos <= 16):
+            if not (0 <= count <= 32 and 1 <= tab_size <= 255 and count * tab_size < 256):
```
Comment (patkan) [inline src/escpos/escpos.py]: This line is a bit convoluted.
Could you maybe split it?
Just a proposal:
```python
count_good = (0 <= count <= 32)
tab_size_good = (1 <= tab_size <= 255)
line_good = (count * tab_size < 256)
if not (count_good and tab_size_good and line_good)
```

Comment (csoft2k) [inline src/escpos/escpos.py]: What about separating each part in a line using no extra variables?
if not (0 <= count <= 32 and
          1 <= tab_size <= 255 and
          count * tab_size < 256):

Comment (patkan) [inline src/escpos/escpos.py]: Sounds also good to me, I am just not 100% sure how to do the indentations :-) But a good editor should help you.

**Pull request merged**